### PR TITLE
Use contents of the script Configmap to generate config hash

### PR DESCRIPTION
**Pull request by cr7258**
Use contents of the script Configmap to generate config hash to avoid unnecessary cluster restart, fix:  https://github.com/elastic/cloud-on-k8s/issues/6963

### CODE REVIEW

**Comment by pebrc** [comment]
buildkite test this -f p=gke,t=TestESSuspendPod

File: pkg/controller/elasticsearch/nodespec/podspec.go [hunk 1]
```diff
@@ -245,3 +246,21 @@ func enableLog4JFormatMsgNoLookups(builder *defaults.PodTemplateBuilder) {
 		}
 	}
 }
+
+// Get contents of the script ConfigMap to generate config hash, excluding the suspended_pods.txt, as it may change over time.
+func getScriptsConfigMapContent(cm *corev1.ConfigMap) string {
```
Comment (pebrc) [inline pkg/controller/elasticsearch/nodespec/podspec.go:251]: Maybe add a unit test for this function?

**Review by pebrc — COMMENTED** [review]
Thanks for the contribution! You change looks good and I think we can merge soon. Could you add a unit test for the `getScriptsConfigMapContent` function?

File: pkg/controller/elasticsearch/nodespec/podspec.go (hunk 1 above)
Comment (cr7258) [reply pkg/controller/elasticsearch/nodespec/podspec.go:251]: Done

**Review by pebrc — APPROVED** [review]

**Comment by pebrc** [comment]
@elasticmachine , run elasticsearch-ci/docs

**Comment by pebrc** [comment]
buildkite test this

**Pull request merged**